aten/src/ATen/native/Pow.cpp

```diff
@@ -28,7 +28,8 @@ Tensor& pow_out(Tensor& result, const Tensor& base, Scalar exp) {
   } else if (!exp.isComplex() && (exp.toDouble() == 1.0)) {
     result.resize_as_(base).copy_(base);
   } else {
-    auto iter = TensorIterator::unary_op(result, base,
+    auto common_dtype = at::result_type(base, exp);
```

Comment:
I think we should update the above condition and also check for 0.0 and 1.0 cases for complex. this was a follow up from the PR merged with some complex related changes before. would be awesome if you could add that :D

Reply:
you just mean removing the `!exp.isComplex()` ? changed it -- let me know if that's correct.

Reply:
no that would be correct when the exponent is complex. All I meant was that for complex exponents = 0. + 0. j and 1. + 0.j, we should do exactly what we do for the case when real exponent is 0.0 or 1.0

Reply:
ah right. I think that will go better in another PR just since it's an unrelated change, I can stack it on this one.

Reply:
great! thanks :D